src/main/java/com/coveo/spillway/trigger/AbstractLimitTrigger.java

```diff
@@ -56,13 +60,26 @@ public AbstractLimitTrigger(LimitTriggerCallback callback) {
    * @return True if the limit is triggered, false otherwise
    */
   protected abstract <T> boolean triggered(
-      T context, int cost, int currentLimitValue, LimitDefinition limitDefinition);
+      T context, int currentLimitValue, LimitDefinition limitDefinition);
 
   @Override
   public <T> void callbackIfRequired(
-      T context, int cost, int currentLimitValue, LimitDefinition limitDefinition) {
-    if (triggered(context, cost, currentLimitValue, limitDefinition)) {
+      T context,
+      int cost,
+      Instant timestamp,
+      int currentLimitValue,
+      LimitDefinition limitDefinition) {
+    Instant currentBucket = calculateBucket(timestamp, limitDefinition.getExpiration());
+
+    if (triggered(context, currentLimitValue, limitDefinition)
```

Comment:
For example your current count is at 98 and you have a callback at the limit capacity of 100, you try to call with a cost of 4. It will trigger the limit and the callback, but the value wont be updated (cause the limit exceeded the capacity). So, next time you do a try call of 4 it will trigger the callback again.